### PR TITLE
[BUG] shapedtw using assert_almost_equal in test_distances

### LINKED ISSUE #765: [BUG] intermittent numeric error on distances test
### Describe the bug

test_distances for shape_dtw fails intermittently. should be fixable by rounding or testing almost equal.

### Steps/Code to reproduce the bug

https://github.com/aeon-toolkit/aeon/actions/runs/6298466015/job/17135313906?pr=758

### Expected results

pass test

### Actual results

``` python-traceback

FAILED aeon/distances/tests/test_distances.py::test_distances[dist11] - AssertionError: assert 7.221098545890[512](https://github.com/aeon-toolkit/aeon/actions/runs/6298466015/job/17135313906?pr=758#step:6:513) == 7.221098545890511
 +  where 7.221098545890511 = compute_distance(array([[ 0.81217268, -0.30587821, -0.26408588, -0.53648431,  0.43270381]]), array([[-0.20837892, -0.02813341, -1.06809805,  0.8201354 , -0.89671779,\n        -0.42087368,  0.25144071, -0.62264404, -0.52897611, -0.45450381]]), metric='shape_dtw')

```

### Versions

_No response_

### PR DESCRIPTION
**Pull request by hadifawaz1999**
fix #765

### CODE REVIEW

**Comment by TonyBagnall** [comment]
as we are here, could you remove the comment at the top

File: aeon/distances/tests/test_distances.py [hunk 1]
```diff
@@ -23,7 +23,9 @@ def _validate_distance_result(
     assert isinstance(dist_result, float)
     assert_almost_equal(dist_result, expected_result)
     assert dist_result == compute_distance(x, y, metric=name)
-    if name != "lcss" and name != "edr":
+    if name == "shape_dtw":
```
Comment (TonyBagnall) [inline aeon/distances/tests/test_distances.py:26]: do we need this extra test? I think 
 assert_almost_equal(dist_result, expected_result)

    if name != "lcss" and name != "edr":
        assert_almost_equal(dist_result, compute_distance(x, y, metric=distance))

is ok?

Also I think you can remove the default value of 10 for expected_results

Comment (TonyBagnall) [reply aeon/distances/tests/test_distances.py:26]: although you might want to set it to 6 dps for safety

**Review by TonyBagnall — CHANGES_REQUESTED** [review]
just some tidying up to do while here

**Comment by chrisholder** [comment]
Ill hop on this branch and remove all my old stuff and comments

File: aeon/distances/tests/test_distances.py (hunk 1 above)
Comment (chrisholder) [reply aeon/distances/tests/test_distances.py]: I removed the redundent if checks here for certain distances. The reason we can it is to check the passing a callable to distance() function will yield the same result.

**Review by TonyBagnall — APPROVED** [review]

**Pull request merged**